zigpy_cc/api.py

```diff
@@ -97,17 +97,50 @@ def close(self):
         frame = obj.to_unpi_frame()
 
         if obj.type == CommandType.SREQ:
-            timeout = Timeouts.SREQ
-            waiter = self.wait_for(
-                CommandType.SRSP, obj.subsystem, obj.command, {}, timeout
-            )
             self._uart.send(frame)
+
+            waiter = self.wait_for(CommandType.SRSP, obj.subsystem, obj.command, {}, Timeouts.SREQ)
             result = await waiter.wait()
-            if (
-                result
-                and "status" in result.payload
-                and result.payload["status"] not in expectedStatus
-            ):
+
+            if obj.command == "dataRequest":
+                payload = {
+                    "endpoint": obj.payload["destendpoint"],
+                    "transid": obj.payload['transid'],
+                }
+
+                waiter = self.wait_for(CommandType.AREQ, Subsystem.AF, "dataConfirm", payload)
+                result = await waiter.wait()
+                LOGGER.debug('Got a dataRequest response: %s', result)
+
+                status = result.payload['status']
+
+                if status != 0:
+                    # No route found
+                    if status == 205:
+                        BEACON_MAX_DEPTH = 0x0F
+                        DEF_NWK_RADIUS = (2 * BEACON_MAX_DEPTH)
+
+                        LOGGER.warning('Received a routing failure! Attempting to discover a new route before failing...')
+
+                        #await self.request(Subsystem.ZDO, 'extRouteDisc', {
+                        #    'dstAddr': obj.payload['dstaddr'],
+                        #    'options': 1,
+                        #    'radius': DEF_NWK_RADIUS
+                        #})
+
+                        await self.request(Subsystem.ZDO, 'extRouteDisc', {
```

Comment:
AFAIK this command is only available on zStack 3.x.0 (3.0.x fails and 1.2 doesn't support it).

Reply:
Thanks for the info. I'll take a look at the older versions as well.

Reply:
I was wrong, it works on those version but make sure you use the following logic: https://github.com/Koenkk/zigbee-herdsman/blob/master/src/adapter/z-stack/adapter/zStackAdapter.ts#L191